Osoitetyokalu.py

```diff
@@ -772,107 +772,48 @@ def set_popup_text(self, dlg, vkm_feature):
-                dlg.TielineEdit.setText(tie)
-                dlg.AjoratalineEdit.setText(ajorata)
-                dlg.OsalineEdit.setText(osa)
-                dlg.EtaisyyslineEdit.setText(etaisyys)
-                dlg.HaLulineEdit.setText(hallinnollinen_luokka)
-                dlg.KuntanimilineEdit.setText(kuntanimi)
-                dlg.KatunimilineEdit.setText(katunimi)
-                dlg.KatunumerolineEdit.setText(katunumero)
-                dlg.ElynimilineEdit.setText(elynimi)
-                dlg.UaluenimilineEdit.setText(ualuenimi)
-                dlg.MaakuntanimilineEdit.setText(maakuntanimi)
-                dlg.Kmtk_idlineEdit.setText(kmtk_id)
-                dlg.M_arvolineEdit.setText(m_arvo)
-
-                road_address = f'{tie}/{ajorata}/{osa}/{etaisyys}'
-
-                return road_address
-
-            except KeyError:
-                point_x = vkm_feature['properties']['x']
-                point_y = vkm_feature['properties']['y']
-                tie = str(vkm_feature['properties']['tie'])
-                ajorata = str(vkm_feature['properties']['ajorata'])
-                osa = str(vkm_feature['properties']['osa'])
+            dlg.TielineEdit.setText(str(vkm_feature['properties']['tie']))
+            dlg.AjoratalineEdit.setText(str(vkm_feature['properties']['ajorata']))
+            dlg.OsalineEdit.setText(str(vkm_feature['properties']['osa']))
+            dlg.EtaisyyslineEdit.setText(str(vkm_feature['properties']['etaisyys']))
+
+        if 'ualuenimi' in vkm_feature['properties']:
+            dlg.UaluenimilineEdit.setText(str(vkm_feature['properties']['ualuenimi']))
+
+        #getting road coordinates and road address that are nearest to the mouse click
+        dlg.XlineEdit.setText(str(vkm_feature['properties']['x']))
+        dlg.YlineEdit.setText(str(vkm_feature['properties']['y']))
+        dlg.HaLulineEdit.setText(str(vkm_feature['properties']['hallinnollinen_luokka']))
+        dlg.KuntanimilineEdit.setText(str(vkm_feature['properties']['kuntanimi']))
+        dlg.ElynimilineEdit.setText(str(vkm_feature['properties']['elynimi']))
+        dlg.MaakuntanimilineEdit.setText(str(vkm_feature['properties']['maakuntanimi']))
+        dlg.Kmtk_idlineEdit.setText(str(vkm_feature['properties']['kmtk_id']))
+        dlg.M_arvolineEdit.setText(str(vkm_feature['properties']['m_arvo']))
+
+        if 'tie' in vkm_feature['properties']:
+            return '{tie}/{ajorata}/{osa}/{etaisyys}'.format(
+                tie = str(vkm_feature['properties']['tie']),
+                ajorata = str(vkm_feature['properties']['ajorata']),
+                osa = str(vkm_feature['properties']['osa']),
                 etaisyys = str(vkm_feature['properties']['etaisyys'])
-                hallinnollinen_luokka = str(vkm_feature['properties']['hallinnollinen_luokka'])
+            )
+        elif 'tie' not in vkm_feature['properties'] and 'katunumero' in vkm_feature['properties']:
```

Comment:
Jos haluat tämän voi muuttaa muotoon `if 'katunumero' in vkm_feature['properties']` koska tien olemassaolo on jo tarkoitettu edellisessä if lauseessa. Myös elsen voi poistaa, koska molemmissa if lauseissa on return, joten koodi siirtyy elsen ehtoon joka tapauksessa, mikäli if lauseet ei toteudu.